src/xregexp.js

```diff
@@ -238,6 +238,14 @@ function getContextualTokenSeparator(match, scope, flags) {
         // No need to separate tokens if at the beginning or end of a group
         match.input[match.index - 1] === '(' ||
         match.input[match.index + match[0].length] === ')' ||
+        // No need to separate tokens if at the beginning of a non-capturing group
+        match.input.slice(match.index - 3, 3) === '(?:' ||
+        // No need to separate tokens if before or after a `|`
+        match.input[match.index - 1] === '|' ||
+        match.input[match.index + match[0].length] === '|' ||
+        // No need to separate tokens if at the beginning or end of the pattern
+        match.input[match.index - 1] === undefined ||
```

Comment:
Can you just check the value of `match.index - 1` directly, without involving `match.input`?

Reply:
Good point, fixed in https://github.com/slevithan/xregexp/pull/196/commits/ee766c211250cc4c90d73123ff1d6095007f0c96 and https://github.com/slevithan/xregexp/pull/196/commits/222eda9017dc424a18b9ff3c26cb04d71946f30d